### PR TITLE
New version of device  TS0601 _TZE200_lsanae15

### PR DESCRIPTION
**Pull request by prokudin07**
Here is [discussion](https://github.com/Koenkk/zigbee2mqtt/discussions/13663)
Device: DAC2161C
image
Did exposes, but for DP17 - only read, although in Tuya app  it's settable:
image

### CODE REVIEW

File: devices/tuya.js [hunk 1]
```diff
@@ -2496,6 +2496,58 @@ module.exports = [
...
+            exposes.text('threshold1_description', ea.STATE),
+            exposes.text('threshold1_value', ea.STATE)
+                .withDescription('Setup value on the device'),
+            exposes.text('threshold2_state', ea.STATE)
+                .withDescription('null - not set, OFF - just alarm, ON - relay will be off when threshold reached'),
+            exposes.text('threshold2_description', ea.STATE),
+            exposes.text('threshold2_value', ea.STATE)
+                .withDescription('Setup value on the device'),
+            exposes.binary('clear_event', ea.STATE_SET, 'ON', 'OFF')
+                .withDescription('Clear event'),
```
Comment (Koenkk) [inline devices/tuya.js:2527]: Does this clear the fault? If yes add this to the description and remove to `clear_fault`

File: lib/tuya.js [hunk 2]
```diff
@@ -1258,6 +1258,37 @@ const valueConverter = {
...
+                power: (buf[7] | buf[6] << 8)};
+        },
+    },
+    alarm: {
+        from: (v) => {
+            const buffer = Buffer.from(v, 'base64');
+            const lookup = {
+                0: 'not set',
+                1: 'Over current threshold',
+                3: 'Over voltage threshold',
```
Comment (Koenkk) [inline lib/tuya.js:1276]: Please make this values lower & snake case, e.g. `over_voltage_threshold`

File: devices/tuya.js [hunk 3]
```diff
@@ -2496,6 +2496,58 @@ module.exports = [
...
+        exposes: [
+            tuya.exposes.switch(),
+            e.energy(),
+            e.power(),
+            e.voltage(),
+            e.current(),
+            exposes.enum('fault', ea.STATE, ['clear', 'over current threshold', 'over power threshold',
+                'over voltage threshold', 'wrong frequency threshold'])
+                .withDescription('Text of fault'),
+            exposes.text('threshold1_state', ea.STATE)
```
Comment (Koenkk) [inline devices/tuya.js:2516]: `thresholdX_state` can become an `exposes.binary()` (this make it easier to use in automations)

File: devices/tuya.js [hunk 4]
```diff
@@ -2496,6 +2496,58 @@ module.exports = [
...
+                .withDescription('null - not set, OFF - just alarm, ON - relay will be off when threshold reached'),
+            exposes.text('threshold1_description', ea.STATE),
```
Comment (Koenkk) [inline devices/tuya.js:2518]: This can become an `exposes.enum`

File: devices/tuya.js [hunk 5]
```diff
@@ -2496,6 +2496,58 @@ module.exports = [
...
+            exposes.text('threshold1_value', ea.STATE)
```
Comment (Koenkk) [inline devices/tuya.js:2519]: I would expect this can become an `exposes.numeric`?

File: devices/tuya.js (hunk 1 above)
Comment (prokudin07) [reply devices/tuya.js]: did it

File: lib/tuya.js (hunk 2 above)
Comment (prokudin07) [reply lib/tuya.js]: did it

File: devices/tuya.js (hunk 3 above)
Comment (prokudin07) [reply devices/tuya.js]: did it

File: devices/tuya.js (hunk 4 above)
Comment (prokudin07) [reply devices/tuya.js:2518]: did it binary

File: devices/tuya.js (hunk 5 above)
Comment (prokudin07) [reply devices/tuya.js]: did it

**Comment by prokudin07** [comment]
Did all recommendations, but can not resolve conflict

File: devices/tuya.js [hunk 6]
```diff
@@ -2496,6 +2496,58 @@ module.exports = [
...
+            exposes.binary('threshold1_state', ea.STATE, 'ON', 'OFF')
+                .withDescription('null - not set, OFF - just alarm, ON - relay will be off when threshold reached'),
+            exposes.text('threshold1_description', ea.STATE),
+            exposes.numeric('threshold1_value', ea.STATE)
+                .withDescription('Setup value on the device'),
+            exposes.binary('threshold2_state', ea.STATE, 'ON', 'OFF')
+                .withDescription('null - not set, OFF - just alarm, ON - relay will be off when threshold reached'),
+            exposes.text('threshold2_description', ea.STATE),
+            exposes.numeric('threshold2_value', ea.STATE)
+                .withDescription('Setup value on the device'),
```
Comment (Koenkk) [inline devices/tuya.js:2525]: What is the unit of this value? add with with `withUnit('theunit')`

File: devices/tuya.js [hunk 7]
```diff
@@ -2496,6 +2496,58 @@ module.exports = [
...
+            exposes.text('meterid', ea.STATE).withDescription('Meter ID'),
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [1, 'energy', tuya.valueConverter.divideBy100],
+                [3, null, null], // Monthly, but sends data only after request
+                [4, null, null], // Dayly, but sends data only after request
+                [6, null, tuya.valueConverter.phaseB], // voltage and current
+                [10, 'fault', tuya.valueConverterBasic.lookup({'clear': 0, 'over current threshold': 1,
+                    'over power threshold': 2, 'over voltage threshold': 4, 'wrong frequency threshold': 8})],
```
Comment (Koenkk) [inline devices/tuya.js:2537]: Please make these snake case, e.g. `over power threshold` -> `over_power_threshold`

File: devices/tuya.js [hunk 8]
```diff
@@ -2496,6 +2496,58 @@ module.exports = [
...
+            exposes.numeric('threshold1_value', ea.STATE)
+                .withDescription('Setup value on the device'),
+            exposes.binary('threshold2_state', ea.STATE, 'ON', 'OFF')
+                .withDescription('null - not set, OFF - just alarm, ON - relay will be off when threshold reached'),
+            exposes.text('threshold2_description', ea.STATE),
+            exposes.numeric('threshold2_value', ea.STATE)
+                .withDescription('Setup value on the device'),
+            exposes.binary('clear_fault', ea.STATE_SET, 'ON', 'OFF')
+                .withDescription('Clear the fault'),
+            exposes.text('meterid', ea.STATE).withDescription('Meter ID'),
```
[the remark this answers is not in the document]
Comment (prokudin07) [reply devices/tuya.js:2528]: did it

File: devices/tuya.js (hunk 7 above)
Comment (prokudin07) [reply devices/tuya.js]: did it

File: devices/tuya.js (hunk 6 above)
Comment (Koenkk) [reply devices/tuya.js:2525]: I see, then and no unit but mention it in the description

Comment (prokudin07) [reply devices/tuya.js:2525]: Did it

File: devices/tuya.js [hunk 9]
```diff
@@ -2496,6 +2496,51 @@ module.exports = [
...
+            exposes.numeric('threshold_1_value', ea.STATE)
+                .withDescription('Can be in Volt or Ampere depending on threshold setting. Setup the value on the device'),
+            exposes.binary('threshold_2_state', ea.STATE, 'ON', 'OFF')
+                .withDescription('OFF - alarm only, ON - relay will be off when threshold reached'),
+            exposes.text('threshold_2_description', ea.STATE),
+            exposes.numeric('threshold_2_value', ea.STATE)
+                .withDescription('Setup value on the device'),
+            exposes.binary('clear_fault', ea.STATE_SET, 'ON', 'OFF')
+                .withDescription('Turn ON to clear last the fault'),
+            exposes.text('meterid', ea.STATE).withDescription('Meter ID'), // TODO: update description
```
Comment (Koenkk) [inline devices/tuya.js:2522]: Can you add a better description here?

File: devices/tuya.js [hunk 10]
```diff
@@ -2496,6 +2496,51 @@ module.exports = [
...
+            exposes.enum('fault', ea.STATE, ['clear', 'over_current_threshold', 'over_power_threshold',
+                'over_voltage threshold', 'wrong_frequency_threshold']).withDescription('Fault status of the device (clear = nothing)'),
+            exposes.binary('threshold_1_state', ea.STATE, 'ON', 'OFF')
+                .withDescription('OFF - alarm only, ON - relay will be off when threshold reached'),
+            exposes.text('threshold_1_description', ea.STATE), // TODO: what is this?
+            exposes.numeric('threshold_1_value', ea.STATE)
+                .withDescription('Can be in Volt or Ampere depending on threshold setting. Setup the value on the device'),
+            exposes.binary('threshold_2_state', ea.STATE, 'ON', 'OFF')
+                .withDescription('OFF - alarm only, ON - relay will be off when threshold reached'),
+            exposes.text('threshold_2_description', ea.STATE),
```
Comment (Koenkk) [inline devices/tuya.js:2517]: Can you add a description for `threshold_2_description`/`threshold_1_description`, what is this?

Comment (Koenkk) [reply devices/tuya.js:2517]: Found it out, please ignore this.

File: devices/tuya.js (hunk 9 above)
Comment (prokudin07) [reply devices/tuya.js]: It's just ID of device. Did it
telegram-cloud-photo-size-2-5449377471087623675-y

**Comment by prokudin07** [comment]
I see your edits, thanks, I'll keep it in mind for the future.

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**